### PR TITLE
supernova: Unify scsynth/supernova plugin loading

### LINKED ISSUE #2290: supernova does not load the same plugin paths as scsynth
supernova isn't working at all for me. I'm on Ubuntu Studio 16.04, 64-bit, latest master.

This code:

```
{ SinOsc.ar }.play;
```

results in this:

```
Exception when reading synthdef: Cannot load synth temp__0: Unit generator Control not installed
Cannot create synth (synthdef: temp__0, node id: 1001)
```

I also get many similar messages for all the synthdefs created by quarks in my class library.

Seems that supernova can't find the server plugin shared object files. It's probably because they're in an unusual place -- my CMAKE_INSTALL_PREFIX is `~/usr/local/`. But how come scsynth finds them fine?

### PR DESCRIPTION
**Pull request by snappizz**
this fixes #2290 and #3446. here are the supernova plugin path problems:

- the `SC_PLUGIN_PATH` environment variable is not respected
- on linux systems, a fixed set of hardcoded paths are tested, and that's it
- tries *all* files, not just ones with the correct extension

now supernova should have identical behavior in plugin path loading compared to scsynth. #2290 in particular was barring me from using supernova.

some variables had to be moved from scsynth to avoid code duplication.

### CODE REVIEW

File: server/supernova/server/main.cpp [hunk 1]
```diff
@@ -242,20 +242,19 @@ void set_plugin_paths(server_arguments const & args, nova::sc_ugen_factory * fac
...
+#ifdef SC_PLUGIN_DIR
+        factory->load_plugin_folder(SC_PLUGIN_DIR);
 #endif
+        factory->load_plugin_folder(SC_Filesystem::instance().getDirectory(DirName::Resource) / SC_PLUGIN_DIR_NAME);
+        factory->load_plugin_folder(SC_Filesystem::instance().getDirectory(DirName::SystemExtension));
+        factory->load_plugin_folder(SC_Filesystem::instance().getDirectory(DirName::UserExtension));
+        vector<std::string> directories;
+        const char * env_plugin_path = getenv("SC_PLUGIN_PATH");
+        if (env_plugin_path) {
+            for (string const & path : directories) {
```
Comment (brianlheim) [inline server/supernova/server/main.cpp]: this always iterates over an empty vector; what was intended here?

Comment (brianlheim) [reply server/supernova/server/main.cpp]: are you trying to iterate over a list of paths in `SC_PLUGIN_PATH`?

Comment (snappizz) [reply server/supernova/server/main.cpp]: heauguuauugh bleauaueuauugh

File: server/supernova/server/main.cpp [hunk 2]
```diff
@@ -242,20 +242,20 @@ void set_plugin_paths(server_arguments const & args, nova::sc_ugen_factory * fac
...
-        factory->load_plugin_folder(SC_Filesystem::instance().getDirectory(DirName::Resource) / "plugins");
-        factory->load_plugin_folder(SC_Filesystem::instance().getDirectory(DirName::SystemExtension) );
-        factory->load_plugin_folder(SC_Filesystem::instance().getDirectory(DirName::UserExtension) );
+#ifdef SC_PLUGIN_DIR
+        factory->load_plugin_folder(SC_PLUGIN_DIR);
 #endif
+        factory->load_plugin_folder(SC_Filesystem::instance().getDirectory(DirName::Resource) / SC_PLUGIN_DIR_NAME);
+        factory->load_plugin_folder(SC_Filesystem::instance().getDirectory(DirName::SystemExtension));
+        factory->load_plugin_folder(SC_Filesystem::instance().getDirectory(DirName::UserExtension));
+        vector<std::string> directories;
```
Comment (brianlheim) [inline server/supernova/server/main.cpp]: this could be moved inside the `if` block

**Comment by snappizz** [comment]
hmmm... what would be a good place to document it?

**Comment by brianlheim** [comment]
Literally anywhere. It could be on a wiki page and then we decide how to integrate it later.

**Comment by brianlheim** [comment]
Well, `SC_PLUGIN_PATH` is documented in the Server Guide helpfile. I think that's good enough for now.

**Pull request merged**